File: 구현/5073 삼각형과 세 변.js
const data = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");
data.pop();   // 0 0 0 값 제거
const result = []
for(let el of data) {
  const input = el.split(' ').map(Number);
  input.sort((a, b) => a - b);
  if(input[0] + input[1] <= input[2]) {
    result.push('Invalid');
    continue;
  }
  const set = new Set(input);   // 같은 값은 하나만 저장되는 집합의 성질 이용
  if(set.size === 1) {
    result.push('Equilateral');
  } else if(set.size === 2) {
    result.push('Isosceles');
  } else {
    result.push('Scalene');
  } 
}
console.log(result.join('\n'));

File: 백준/구현/5073 삼각형과 세 변.py
while True:
  triangle = sorted(list(map(int, input().split())))
  if triangle[0] == triangle[1] == triangle[2] == 0:
     break
  if triangle[0] + triangle[1] <= triangle[2]:   # 두 변의 길이가 가장 긴 변의 길이보다 길지 않은 경우  
    print('Invalid')
  elif triangle[0] == triangle[1] == triangle[2]:   # 세 변의 길이가 모두 같은 경우
    print('Equilateral')
  elif triangle[0] == triangle[1] or triangle[1] == triangle[2] or triangle[2] == triangle[0]:   # 두 변의 길이만 같은 경우
    print('Isosceles')
  else:   # 세 변의 길이가 모두 다른 경우
    print('Scalene')
